plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways.tsx

```diff
@@ -14,107 +14,163 @@ import { getAdminLink } from '@woocommerce/settings';
  * Internal dependencies
  */
 
+interface PaymentGateway {
+	id: string;
+	title: string;
+	description: string;
+	order: string | number;
+	enabled: boolean;
+	method_title: string;
+	method_description: string;
+	needs_setup: boolean;
+	image?: string;
+	image_72x72?: string;
+	square_image?: string;
+	actionText?: string;
+	recommended?: boolean;
+}
+
+interface WooPaymentsData {
+	isSupported: boolean;
+	isOnboarded: boolean;
+	isInTestMode: boolean;
+}
+
+const parseScriptTag = ( elementId: string ) => {
+	const scriptTag = document.getElementById( elementId );
+	return scriptTag ? JSON.parse( scriptTag.textContent || '' ) : [];
+};
+
 export const PaymentGateways = () => {
-	// Mock payment gateways for now.
-	// TODO Get the list of gateways via the API in future PR.
-	const mockPaymentGateways = [
-		{
-			id: 'woocommerce_payments',
-			title: __( 'Accept payments with Woo', 'woocommerce' ),
-			content: __(
-				'Credit/debit cards, Apple Pay, Google Pay & more.',
-				'woocommerce'
-			),
-			image: 'https://woocommerce.com/wp-content/plugins/wccom-plugins/payment-gateway-suggestions/images/wcpay.svg',
-			square_image:
-				'https://woocommerce.com/wp-content/plugins/wccom-plugins/payment-gateway-suggestions/images/woocommerce.svg',
-			image_72x72:
-				'https://woocommerce.com/wp-content/plugins/wccom-plugins/payment-gateway-suggestions/images/wcpay.svg',
-			actionText: '',
-			recommended: true,
-		},
-		{
-			id: 'ppcp-gateway',
-			title: 'PayPal Payments',
-			content: __(
-				"Safe and secure payments using credit cards or your customer's PayPal account.",
-				'woocommerce'
-			),
-			image: 'https://woocommerce.com/wp-content/plugins/woocommerce/assets/images/paypal.png',
-			image_72x72:
-				'https://woocommerce.com/wp-content/plugins/wccom-plugins/payment-gateway-suggestions/images/72x72/paypal.png',
-			square_image:
-				'https://woocommerce.com/wp-content/plugins/wccom-plugins/payment-gateway-suggestions/images/paypal.svg',
-			actionText: '',
-			recommended: false,
-		},
-	];
+	const [ paymentGateways, setPaymentGateways ] = useState( [] );
+	const [ wooPaymentsData, setWooPaymentsData ] = useState( {} );
+
+	const recommendedGateways = [ 'woocommerce_payments' ];
+
+	useEffect( () => {
+		setWooPaymentsData(
+			parseScriptTag( 'experimental_wc_settings_payments_woopayments' )
+		);
+		setPaymentGateways(
+			parseScriptTag( 'experimental_wc_settings_payments_gateways' )
+		);
+	}, [] );
+
+	const enableGateway = ( gateway: PaymentGateway ) => () => {
+		console.log( 'Enable gateway', gateway );
```

Comment:
```suggestion
		// TODO implement in future PR
```

Reply:
I'll implement this one in this PR, will update it tomorrow. :)